Source/Dafny/DafnyAst.cs

```diff
@@ -4694,29 +4734,36 @@ public enum ES { NotYetComputed, Never, ConsultTypeArguments }
 
   public class TupleTypeDecl : IndDatatypeDecl
   {
-    public readonly int Dims;
+    public readonly List<bool> GhostArguments;
+
+    public int Dims
```

Comment:
This field is used in `Compiler-java.cs` and I think there it means the runtime dimensions of the tuple type. Consider introduced a `NonGhostDims` field and seeing which usages of `Dims` should use that.

Reply:
I added `NonGhostDims` with a TODO to check the usages of `Dims`.